engine/process_p2p_transaction.go

```
@@ -0,0 +1,101 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/bitcoin-sv/go-paymail"
+)
+
+// processP2PTransaction will process the sync transaction record, or save the failure
+func processP2PTransaction(ctx context.Context, tx *Transaction) error {
+	// Successfully capture any panics, convert to readable string and log the error
+	defer recoverAndLog(tx.Client().Logger())
+
+	syncTx := tx.syncTransaction
+	// Create the lock and set the release for after the function completes
+	unlock, err := newWriteLock(
+		ctx, fmt.Sprintf(lockKeyProcessP2PTx, syncTx.GetID()), syncTx.Client().Cachestore(),
+	)
+	defer unlock()
+	if err != nil {
+		return err
+	}
+
+	// No draft?
+	if len(tx.DraftID) == 0 {
+		syncTx.addSyncResult(ctx, syncActionP2P, "all", "no draft found, cannot complete p2p")
+
+		return nil // TODO: why nil here??
```

Shouldn't we return an error here and handle it "higher"?